Drop support for mac x86 (#1716)
PennyLaneAI/catalyst: doc/dev/installation.rst

    The easiest method of installation is to run ``xcode-select --install`` from the Terminal
    app.

.. note::

>   Catalyst does not currently support macOS with x86_64 architecture. This includes Macs running on Intel processors.

Reviewer: ```suggestion
  Catalyst no longer supports macOS with x86_64 architecture. This includes Macs running on Intel processors.
```

Author: You are right, "does not currently support" sounds like this will be added in the future :laughing:

Author: https://github.com/PennyLaneAI/catalyst/pull/1716/commits/8fe33415d53aa98ab157507fd8b89c64996fbe1a
